### ISSUE TITLE
Embedded Shopfronts - All allowed URLs in header

### ISSUE THREAD

**Issue by lin-d-hop** [openfoodfoundation/openfoodnetwork#1953]
There is a strange error happening in embedded shopfronts in which there appears to be some kind of hierarchical dependency within the list of Allowed URLs recorded in super-admin.

We have the following URLs on our list:
`dev.landworkersalliance.org.uk
www.foodmadegood.org
www.blackmountainfoodhub.co.uk
www.rootsfruitsandleaves.co.uk`

When I add a new shopfront to the list I get the following error in Chrome console when I load the page with the embedded shop. In this case I loaded Black Mountain Food Hub. But the embedded frame is failing apparently due to errors in Food Made Good (different group site).
![lowerlevelembed](https://user-images.githubusercontent.com/8669848/32722107-b7a41dce-c860-11e7-9bfd-531efba09978.png)
Regardless of whether of not there are errors in that site, surely loading www.blackmountainfoodhub.org.uk should know absolutely nothing about www.foodmadegood.org

Note the order of the URLs is important. The top URL in the list will load the embedded shopfront but will display the following errors in Chrome console on page load
![toplevelembed](https://user-images.githubusercontent.com/8669848/32722368-9b4f7820-c861-11e7-8682-4ffd298e468c.png)

This issue is very high priority for UK as it is currently causing shopfronts to fail to load. In particular, playing around trying to understand this issue currently our established embedded shopfront is not loading for the user = big problem for them.

Unfortunately everyone that can help with this issue is on holiday! x-/

**Comment by lin-d-hop** [comment]
Ok, so I spent a while looking into this.
The issue was caused due to very-easy-to-repeat user error by me. But I do think there is grounds to change something here....

So when I entered the Embeddable URLs into production I did so by entering each URL on a new line.
This became, in the page headers something like:
frame-ancestors = www.url1.com /rwww.url2.com /rwww.url3.com

This resulted in the first url being recognisable but all future URLs being deemed as illegal.

So the solution to actually have pages embed was to ensure that a **space** exists so that page headers actually contain:
frame-ancestors = www.url1.com www.url2.com www.url3.com

This did get me wondering, however, if it is a good idea to embed the URL of every that we have enabled shopfronts for in every header? Maybe I'm being over zealous, but it felt kinda intrusive to be able to see from one website the list of all other websites using OFN embeddable shopfronts.

I wonder if it would be better to instead just enable in the header the shopfront that made the request, after checking on our servers that the requesting URI is on the allowed list.

I made a totally untested PR #1954 for this change just to get some feedback if other devs think it is a good idea. I didn't put any time into figuring out how to test with locally so probably doesn't work out of the box.
